src/structured-text/converters/convert_int_value.cpp

```diff
@@ -0,0 +1,27 @@
+/*******************************************************************\
+
+Module: Structured Text Language Conversion
+
+Author: Matthias Weiss, matthias.weiss@diffblue.com
+
+\*******************************************************************/
+
+/// \file
+/// Structured Text Language Conversion
+
+#include "convert_int_value.h"
+#include <util/std_types.h>
+
+constant_exprt convert_int_value(const std::string &src)
+{
+  signedbv_typet type(16);
+  type.set(ID_structured_text_type, ID_structured_text_int_value);
+  return constant_exprt(src, type);
```

Comment:
this seems very error prone as the irep_idt in `constant_exprt` can be interpreted in a different way than the string is (in particular the base may not correspond), I would have expected some conversion from `src` to int to happen and then `from_integer` to be used.

Reply:
STL integer literal syntax doesn't match the one of C. So far, only `123`, `-123` and `+123` appear to be allowed for `INT` types in the STL grammar. I'd suggest at least removing stray `+` characters, which should make them compatible without explicit `std::stoi` call.

However, perhaps add a `TODO` to later introduce regression tests for invalid inputs, in order to check that the grammar properly rejects incorrectly specified literals.

Reply:
Added functions for different Int literal representations. I'll add the regression tests once the assignment of literals is supported by the grammar.